test/store_unittest.cc

```diff
@@ -0,0 +1,226 @@
+  MonitoredResource mr("some_resource", {});
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m));
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_FALSE(metadata_map.empty());
+  EXPECT_EQ("default-version", metadata_map.at(mr).version);
+}
+
+TEST_F(MetadataStoreTest, MultipleMetadataEntries) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m2(
+      "default-version2",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr1, std::move(m1));
+  store.UpdateMetadata(mr2, std::move(m2));
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_FALSE(metadata_map.empty());
+  EXPECT_EQ("default-version1", metadata_map.at(mr1).version);
+  EXPECT_EQ("default-version2", metadata_map.at(mr2).version);
+}
+
+TEST_F(MetadataStoreTest, UpdateMetadataEntry) {
+  MonitoredResource mr("some_resource", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m1));
+  const auto metadata_map_before = GetMetadataMap();
+  EXPECT_FALSE(metadata_map_before.empty());
+  EXPECT_EQ("default-version1", metadata_map_before.at(mr).version);
+  MetadataStore::Metadata m2(
+      "default-version2",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m2));
+  const auto metadata_map_after = GetMetadataMap();
+  EXPECT_EQ("default-version2", metadata_map_after.at(mr).version);
+}
+
+TEST_F(MetadataStoreTest, PurgeDeletedEntries) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m2(
+      "default-version2",
+      true,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr1, std::move(m1));
+  store.UpdateMetadata(mr2, std::move(m2));
+  const auto metadata_map_before = GetMetadataMap();
+  EXPECT_FALSE(metadata_map_before.empty());
```

Comment:
You can check the exact size instead, here and below.

Reply:
Done.